pkg/kmm/module.go

```diff
@@ -359,6 +367,10 @@ func (builder *ModuleBuilder) withServiceAccount(srvAccountName string, accountT
 	case "module":
 		builder.Definition.Spec.ModuleLoader.ServiceAccountName = srvAccountName
 	case "device":
```

Comment:
I would probably define a new type for this:
```go
type ServiceAccountType string

const (
    ServiceAccountTypeModule ServiceAccountType = "module"
    ServiceAccountTypeDevice ServiceAccountType = "device"
)
```
The `switch` then becomes:
```go
	switch accountType {
	case ServiceAccountTypeModule:
		builder.Definition.Spec.ModuleLoader.ServiceAccountName = srvAccountName
	case ServiceAccountTypeDevice:
```

Reply:
I think I will leave it as is for the moment, given that this will change the signature of the function, causing other refactorings